lib/dry/files/file_system.rb

```diff
@@ -117,13 +126,16 @@ def touch(path, **kwargs)
       # All the intermediate directories are created.
       #
       # @param path [String,Pathname] the path to file
-      # @param content [String, Array<String>] the content to write
+      # @param content [String] the content to write
       #
       # @raise [Dry::Files::IOError] in case of I/O error
+      # @raise [CanOnlyWriteStringError] if content param isn't a String
       #
       # @since 0.1.0
       # @api private
-      def write(path, *content)
+      def write(path, content)
+        raise CanOnlyWriteStringError unless content.is_a?(String)
```

Comment:
@cllns Given this is an adapter (private API), we control the arguments, this exception seems to be a redundant internal assertion. Is this necessary?

Reply:
Gotcha. Was being defensive but you're right, it's not worth a custom Error